pkg/reconciler/finalizer.go

```diff
@@ -0,0 +1,65 @@
+package reconciler
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlutil "sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type Finalizer struct {
```

Comment:
Is this finalizer file being used at all during cleanup after Kepler resource has been deleted? From what I have seen, it seems only the deleter.go is being used. Originally, finalizer was only being used for cleanup of resources connected to Kepler Resources.

Reply:
> Is this finalizer file being used at all during cleanup after Kepler resource has been deleted? From what I have seen, it seems only the deleter.go is being used. Originally, finalizer was only being used for cleanup of resources connected to Kepler Resources.

Reply:
Yes, we are using finalizer to ensure all cluster-scoped resources are deleted before deleting kepler